### PR TITLE
Use shape when rendering Http errors to allow cusomization from the UI

### LINKED ISSUE #14548: Overriding Error Pages from the Template feature
### Is your feature request related to a problem? Please describe.
I want to be able to override the error pages using the Template feature. 

### Describe the solution you'd like
Compose the error views as shapes instead of a standard MVC-action view.

### Discussed in https://github.com/OrchardCMS/OrchardCore/discussions/14543

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Change Diagnostic errors to shape to allow us to change the template of each view.
Fix #14548

Also added a missing documentation page for the Diagnostics module.

Now, if you want to customize the 404 pages using Liquid template, Enable "Templates" feature then add a template called `NotFound` and design it how you want it.

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@jtkech can you please review this?

**Comment by jtkech** [comment]
Can't right now, I have other things to do, but will look at it asap

**Comment by MikeAlhayek** [comment]
@jtkech reminder to review this please.

File: src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    public HttpStatusCode HttpStatusCode { get; }
+
+    public HttpStatusCodeShapeViewModel(HttpStatusCode statusCode)
+    {
+        HttpStatusCode = statusCode;
+        Metadata.Type = "HttpStatusCode";
+
+        // The type name is 'HttpStatusCode', so any Http status code will be handled by the 'HttpStatusCode.cshtml' or 'HttpStatusCode.liquid'.
+        // However, assign an alternative for every 'StatusCode' to enable the customization for each Status (ex. 'NotFound.cshtml' or 'NotFound.liquid').
+        Metadata.Alternates.Add(statusCode.ToString());
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs:17]: Maybe add the int value too as an alternate.

File: src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace OrchardCore.Diagnostics.ViewModels;
+
+public class HttpStatusCodeShapeViewModel : ShapeViewModel
+{
+    public HttpStatusCode HttpStatusCode { get; }
+
+    public HttpStatusCodeShapeViewModel(HttpStatusCode statusCode)
+    {
+        HttpStatusCode = statusCode;
+        Metadata.Type = "HttpStatusCode";
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs:13]: Can we have a common type (`Error`, `HttpError`) and use `_NotFound`, `_404` -> `HttpError.404.cshtml` and `HttpError.NotFound.cshtml`

File: src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs (hunk 1 above)
Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs:17]: https://andrewlock.net/netescapades-enumgenerators-a-source-generator-for-enum-performance/

**Review by sebastienros — APPROVED** [review]

File: src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs (hunk 2 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs]: done!

File: src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs (hunk 1 above)
Comment (MikeAlhayek) [reply src/OrchardCore.Modules/OrchardCore.Diagnostics/ViewModels/HttpStatusCodeShapeViewModel.cs]: I added the 404 to the alternate. We'll look at the library and see how we can use it widely but in a separate PR.

**Pull request merged**